### ISSUE TITLE
[gtk3] Runtime failure, due to UB in gtk/gdk/win32/gdkselection-win32.c

### ISSUE THREAD

**Issue by Febbe** [microsoft/vcpkg#30390]
**Describe the bug**
Compiling xournalpp with msvc results in a crash directly after startup.
The crash is located in gobject-2.0-0.dll it crashes with an access violation after a call to `gtk_clipboard_request_contents`.

**Environment**
- OS: Windows
- Compiler: MSVC 143

**To Reproduce**
1. ./vcpkg install gtk3
2. clone xournalpp
3. clone portaudio and compile portaudiocpp, add it to path
4. cd */xournalpp && mkdir build && cd build
5. cmake -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake ..
6. cmake --build .
7. .\xournalpp.exe

**Expected behavior**
No crash with access violation in a 3rd party lib, fetched by vcpkg

**Failure logs**
```
-------------------------------------------------------------------
You may only use the C/C++ Extension for Visual Studio Code
with Visual Studio Code, Visual Studio or Visual Studio for Mac
software to help you develop and test your applications.
-------------------------------------------------------------------
Breakpoint warning: No executable code of the debugger’s target code type is associated with this line.
Possible causes include: conditional compilation, compiler optimizations, or the target architecture of this line is not supported by the current debugger code type. - c:\Users\Febbe\workspace\xournalpp\src\core\gui\toolbarMenubar\PluginToolButton.h:22
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\xournalpp.exe'. Symbols loaded.
Loaded 'C:\Windows\System32\ntdll.dll'. 
Loaded 'C:\Windows\System32\kernel32.dll'. 
Loaded 'C:\Windows\System32\KernelBase.dll'. 
Loaded 'C:\Windows\System32\shell32.dll'. 
Loaded 'C:\Windows\System32\msvcp_win.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\gdk-3-vs17.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\gtk-3-vs17.dll'. 
Loaded 'C:\Windows\System32\ucrtbase.dll'. 
Loaded 'C:\Windows\System32\advapi32.dll'. 
Loaded 'C:\Windows\System32\user32.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\gdk_pixbuf-2.0-0.dll'. 
Loaded 'C:\Windows\System32\msvcrt.dll'. 
Loaded 'C:\Windows\System32\win32u.dll'. 
Loaded 'C:\Windows\System32\sechost.dll'. 
Loaded 'C:\Windows\System32\gdi32.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\gobject-2.0-0.dll'. 
Loaded 'C:\Windows\System32\gdi32full.dll'. 
Loaded 'C:\Windows\System32\rpcrt4.dll'. 
Loaded 'C:\Windows\System32\imm32.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\glib-2.0-0.dll'. 
Loaded 'C:\Windows\System32\ole32.dll'. 
Loaded 'C:\Windows\System32\setupapi.dll'. 
Loaded 'C:\Windows\System32\ws2_32.dll'. 
Loaded 'C:\Windows\System32\combase.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\intl-8.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\cairo-2.dll'. 
Loaded 'C:\Windows\System32\comdlg32.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\pango-1.0-0.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\cairo-gobject-2.dll'. 
Loaded 'C:\Windows\System32\SHCore.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\epoxy-0.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\gio-2.0-0.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\pangocairo-1.0-0.dll'. 

Loaded 'C:\Windows\System32\shlwapi.dll'. 
Unloaded 'C:\Windows\System32\shlwapi.dll'.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\libpng16.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\jpeg62.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\tiff.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\gmodule-2.0-0.dll'. 
Loaded 'C:\Windows\System32\vcruntime140.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\harfbuzz.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\fribidi-0.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\atk-1.0-0.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\pangowin32-1.0-0.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\libffi.dll'. 
Loaded 'C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.22621.608_none_a9444ca7c10bb01d\comctl32.dll'. 
Loaded 'C:\Windows\System32\winspool.drv'. 
Loaded 'C:\Windows\System32\dwmapi.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\poppler-glib.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\zip.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\libxml2.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\portaudio.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\zlib1.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\sndfile.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\iconv-2.dll'. 
Loaded 'C:\Windows\System32\msvcp140.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\pcre2-8.dll'. 
Loaded 'C:\Windows\System32\vcruntime140_1.dll'. 
Loaded 'C:\Windows\System32\winmm.dll'. 
Loaded 'C:\Windows\System32\hid.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\fontconfig-1.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\freetype.dll'. Symbols loaded.
Loaded 'C:\Windows\System32\msimg32.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\pixman-1-0.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\pangoft2-1.0-0.dll'. 
Loaded 'C:\Windows\System32\dnsapi.dll'. 
Loaded 'C:\Windows\System32\IPHLPAPI.DLL'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\liblzma.dll'. 
Loaded 'C:\Windows\System32\DWrite.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\bz2.dll'. 
Loaded 'C:\Windows\System32\bcrypt.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\poppler.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\vorbisenc.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\FLAC.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\opus.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\mpg123.dll'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\libmp3lame.DLL'. 
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\vorbis.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\ogg.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\libexpat.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\brotlidec.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\openjp2.dll'. Symbols loaded.
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\brotlicommon.dll'. Symbols loaded.
Loaded 'C:\Windows\System32\nsi.dll'. 
Loaded 'C:\Windows\System32\bcryptprimitives.dll'. 
Loaded 'C:\Windows\System32\kernel.appcore.dll'. 

(xournalpp.exe:20164): GLib-GIO-WARNING **: 01:08:35.395: win32 session dbus binary not found
Loaded 'C:\Windows\System32\KBDGR.DLL'. 
Loaded 'C:\Windows\System32\devobj.dll'. 
2
Loaded 'C:\Windows\System32\cfgmgr32.dll'. 
Unloaded 'C:\Windows\System32\cfgmgr32.dll'.
Loaded 'C:\Windows\System32\cfgmgr32.dll'. 
Unloaded 'C:\Windows\System32\cfgmgr32.dll'.
Loaded 'C:\Windows\System32\wintrust.dll'. 
Loaded 'C:\Windows\System32\crypt32.dll'. 
Loaded 'C:\Windows\System32\msasn1.dll'. 
Loaded 'C:\Windows\System32\uxtheme.dll'. 
Loaded 'C:\Windows\System32\clbcatq.dll'. 
Loaded 'C:\Windows\System32\msctf.dll'. 
Loaded 'C:\Windows\System32\windows.storage.dll'. 
Loaded 'C:\Windows\System32\WinTypes.dll'. 
Loaded 'C:\Windows\System32\profapi.dll'. 
Loaded 'C:\Windows\System32\winhttp.dll'. 
Loaded 'C:\Windows\System32\OnDemandConnRouteHelper.dll'. 
Unloaded 'C:\Windows\System32\OnDemandConnRouteHelper.dll'.
** Message: 01:08:35.443: TEXTDOMAINDIR = (NULL), Platform-specific locale dir = C:\Users\Febbe\workspace\xournalpp\build\..\share\xournalpp\..\locale, chosen directory = C:\Users\Febbe\workspace\xournalpp\build\..\share\xournalpp\..\locale
Loaded 'C:\Windows\System32\winmmbase.dll'. 
Loaded 'C:\Windows\System32\MMDevAPI.dll'. 
Loaded 'C:\Windows\System32\wdmaud.drv'. 
Loaded 'C:\Windows\System32\ksuser.dll'. 
Loaded 'C:\Windows\System32\avrt.dll'. 
Loaded 'C:\Windows\System32\AudioSes.dll'. 
Loaded 'C:\Windows\System32\msacm32.drv'. 
Loaded 'C:\Windows\System32\msacm32.dll'. 
Loaded 'C:\Windows\System32\midimap.dll'. 
Loaded 'C:\Windows\System32\ResourcePolicyClient.dll'. 
Loaded 'C:\Windows\System32\dsound.dll'. 
Loaded 'C:\Windows\System32\RESAMPLEDMO.DLL'. 
2
Loaded 'C:\Windows\System32\powrprof.dll'. 
Unloaded 'C:\Windows\System32\powrprof.dll'.
Loaded 'C:\Windows\System32\oleaut32.dll'. 
Loaded 'C:\Windows\System32\msdmo.dll'. 
Loaded 'C:\Windows\System32\umpdc.dll'. 

(xournalpp.exe:20164): Gtk-WARNING **: 01:08:35.551: Could not load image 'pixmaps/com.github.xournalpp.xournalpp.svg': Couldn?t recognize the image file format for file ?C:\Users\Febbe\workspace\xournalpp\ui\pixmaps/com.github.xournalpp.xournalpp.svg?
Loaded 'C:\Windows\System32\shfolder.dll'. 
Fontconfig error: Cannot load default config file: No such file: (null)
Loaded 'C:\Windows\System32\cryptbase.dll'. 
Loaded 'C:\Windows\System32\IconCodecService.dll'. 
Loaded 'C:\Windows\System32\WindowsCodecs.dll'. 
Loaded 'C:\Windows\System32\xmllite.dll'. 
Loaded 'C:\Windows\System32\AppXDeploymentClient.dll'. 
Loaded 'C:\Windows\System32\Windows.ApplicationModel.dll'. 
Loaded 'C:\Windows\System32\Windows.StateRepositoryPS.dll'. 
Loaded 'C:\Windows\System32\Windows.StateRepositoryBroker.dll'. 
Loaded 'C:\Windows\System32\MrmCoreR.dll'. 
Loaded 'C:\Windows\System32\iertutil.dll'. 
Loaded 'C:\Windows\System32\Windows.StateRepositoryCore.dll'. 
Loaded 'C:\Windows\System32\Windows.UI.dll'. 
Loaded 'C:\Windows\System32\BCP47mrm.dll'. 
Loaded 'C:\Windows\System32\Windows.FileExplorer.Common.dll'. 
Loaded 'C:\Windows\System32\propsys.dll'. 
Loaded 'C:\Windows\System32\apphelp.dll'. 
Loaded 'C:\Windows\System32\networkexplorer.dll'. 
onecore\com\combase\objact\objact.cxx(4070)\combase.dll!00007FFCE4925905: (caller: 00007FFCE492D65A) ReturnHr(1) tid(4b68) 80040154 Klasse nicht registriert
Loaded 'C:\Windows\System32\mssprxy.dll'. 
Loaded 'C:\Windows\System32\Windows.StateRepositoryClient.dll'. 
onecore\com\combase\objact\objact.cxx(4070)\combase.dll!00007FFCE4925905: (caller: 00007FFCE492D65A) ReturnHr(2) tid(4b68) 80040154 Klasse nicht registriert
onecore\com\combase\objact\objact.cxx(4070)\combase.dll!00007FFCE4925905: (caller: 00007FFCE492D65A) ReturnHr(3) tid(4b68) 80040154 Klasse nicht registriert
onecore\com\combase\objact\objact.cxx(4070)\combase.dll!00007FFCE4925905: (caller: 00007FFCE492D65A) ReturnHr(4) tid(4b68) 80040154 Klasse nicht registriert
Exception thrown at 0x00007FFC855B2975 (gobject-2.0-0.dll) in xournalpp.exe: 0xC0000005: Access violation reading location 0xFFFFFFFFFFFFFFFF.
The program '[20164] xournalpp.exe' has exited with code 0 (0x0).
```

**Additional context**

**Comment by Febbe** [comment]
@JonLiu1993 the problem is, that xournalpp depends on portaudiocpp, which is currently not part of vcpkg, therefore it must be compiled manually: I've just created a static library (release mode + /MD + MSVC STL) and added it to the PATH variable of the OS.

You must install it into `C:\builds` or you must change the prefix path in the "lib\pkgconfig\portaudiocpp.pc" file.

**Comment by JonLiu1993** [comment]
> @JonLiu1993 the problem is, that xournalpp depends on portaudiocpp, which is currently not part of vcpkg, therefore it must be compiled manually: I've just created a static library (release mode + /MD + MSVC STL) and added it to the PATH variable of the OS. I've also added the pkg-config folderpath to the PKG_CONFIG_PATH environment variable.
> 
> If that's too much effort or too complicated, that's my compiled portaudio (release mode + /MD + MSVC STL + x86_64): [portaudiocpp.zip](https://github.com/microsoft/vcpkg/files/11086764/portaudiocpp.zip) You must install it into `C:\builds` or you must change the prefix path in the `lib\pkgconfig\portaudiocpp.pc` file.
> 
> Interestingly, linking against the UCRT64 libraries of MSYS (requires some work to make the pkg-config files compatible), does not result into a crash at runtime. But scrolling/zooming just don't work.

@Febbe, Sorry for reply you so late, could you elaborate on your steps to reproduce it? I couldn't reproduce your issue locally because I really didn't understand your steps to reproduce.

**Comment by Febbe** [comment]
> > @JonLiu1993 the problem is, that xournalpp depends on portaudiocpp, which is currently not part of vcpkg, therefore it must be compiled manually: I've just created a static library (release mode + /MD + MSVC STL) and added it to the PATH variable of the OS. I've also added the pkg-config folderpath to the PKG_CONFIG_PATH environment variable.
> > If that's too much effort or too complicated, that's my compiled portaudio (release mode + /MD + MSVC STL + x86_64): [portaudiocpp.zip](https://github.com/microsoft/vcpkg/files/11086764/portaudiocpp.zip) You must install it into `C:\builds` or you must change the prefix path in the `lib\pkgconfig\portaudiocpp.pc` file.
> > Interestingly, linking against the UCRT64 libraries of MSYS (requires some work to make the pkg-config files compatible), does not result into a crash at runtime. But scrolling/zooming just don't work.
> 
> @Febbe, Sorry for reply you so late, could you elaborate on your steps to reproduce it? I couldn't reproduce your issue locally because I really didn't understand your steps to reproduce.

Which step you don't understand, or at which step do you fail?
Do you have a cmake log/trace? MAybe a specific package is missing.


1.    ./vcpkg install gtk3
2.    clone xournalpp
3.    clone portaudio and compile portaudiocpp, add it to path
4.    cd */xournalpp && mkdir build && cd build
6.    cmake -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake ..
7.    cmake --build .
8.    .\xournalpp.exe

We first need to install all dependencies of xournalpp:
```
glib-2.0
gtk+-3.0
poppler-glib
gthread-2.0
libxml-2.0
libzip
portaudiocpp
sndfile
librsvg-2.0
zlib
imagemagick
```

`./vcpkg install gtk3:x64-windows glib:x64-windows poppler[glib]:x64-windows portaudio:x64-windows libzip:x64-windows zlib:x64-windows libxml2:x64-windows sndfile:x64-windows librsvg:x64-windows libsndfile:x64-windows`

ImageMagick must be acquired from somewhere else. Also, the legacy tools are required to be installed and added to the `PATH`

Then we can choose any working directory and clone xournalpp.
Since Xournalpp requires portaudiocpp, which is currently not part of vcpkg, you must compile it yourself, or use my provided binary.

Now xournalpp should be buildable as described by step 4 and onwards.

**Comment by dg0yt** [comment]
> ImageMagick must be acquired from somewhere else. Also, the legacy tools are required to be installed and added to the `PATH`

Wait, this will pull in non-vcpkg builds of graphic libraries and font libraries. It is no surprise that this gives runtime errors.

And of course one must not mix ucrt and legacy crt binaries.

**Comment by Febbe** [comment]
> > ImageMagick must be acquired from somewhere else. Also, the legacy tools are required to be installed and added to the `PATH`
> 
> Wait, this will pull in non-vcpkg builds of graphic libraries and font libraries. It is no surprise that this gives runtime errors.
> 
> And of course one must not mix ucrt and legacy crt binaries.

Wouldn't the log show that there is a library loaded from such a location?
I can only find some from the system path and the build path. So that shouldn't be the issue here, right? I can rewrite the cmake build script through, that it requires the path in a cmake variable to imagemagic, instead of having it in the path variable.

**Comment by dg0yt** [comment]
Which DLLs come with your copy of imagemagick? (Are there any DLLs at all?)

**Comment by Febbe** [comment]
```
"msvcp140_codecvt_ids.dll"
"vcomp140.dll"
"vcruntime140.dll"
"vcruntime140_1.dll"
"CORE_RL_bzlib_.dll"
"CORE_RL_cairo_.dll"
"CORE_RL_exr_.dll"
"CORE_RL_flif_.dll"
"CORE_RL_freetype_.dll"
"CORE_RL_fribidi_.dll"
"CORE_RL_glib_.dll"
"CORE_RL_harfbuzz_.dll"
"CORE_RL_heif_.dll"
"CORE_RL_jasper_.dll"
"CORE_RL_jpeg-turbo_.dll"
"CORE_RL_jpeg-xl_.dll"
"CORE_RL_lcms_.dll"
"CORE_RL_lqr_.dll"
"CORE_RL_lzma_.dll"
"CORE_RL_Magick++_.dll"
"CORE_RL_MagickCore_.dll"
"CORE_RL_MagickWand_.dll"
"CORE_RL_openjpeg_.dll"
"CORE_RL_pango_.dll"
"CORE_RL_png_.dll"
"CORE_RL_raw_.dll"
"CORE_RL_rsvg_.dll"
"CORE_RL_tiff_.dll"
"CORE_RL_webp_.dll"
"CORE_RL_xml_.dll"
"CORE_RL_zip_.dll"
"CORE_RL_zlib_.dll"
"mfc140u.dll"
"msvcp140.dll"
"msvcp140_1.dll"
"msvcp140_2.dll"
"msvcp140_atomic_wait.dll"
```

**Comment by Febbe** [comment]
Ok, I found another issue: rsvg-convert.exe is not builded and installed with the librsvg target, so you musst download it from: [here](https://sourceforge.net/projects/tumagcc/files/converters/rsvg-convert.exe/download) and put it into the path.
Also, pkg-connfig is not installed along with vcpkg, which also musst be aquired from somewhere, I took that: https://sourceforge.net/projects/pkgconfiglite/files/0.28-1/pkg-config-lite-0.28-1.tar.gz/download

Also, we musst create a pc file at `C:\vcpkg\installed\x64-windows\lib\pkgconfig\librsvg-2.0.pc` with the following content:
```
prefix=${pcfiledir}/../..
exec_prefix=${prefix}
bindir=${prefix}/bin
libdir=${prefix}/lib
includedir=${prefix}/include

Name: librsvg-2.0
Description: library for handling rsvg files
Version: 2.40.2
Requires: glib-2.0 gio-2.0 gdk-pixbuf-2.0 cairo
Libs:  "-L${libdir}" -lrsvg-2.40
Cflags: "-I${includedir}"
```

**Comment by Febbe** [comment]
So I added stacktraces for windows builds and took care, that no foreign dll is in the path (possible from pkg-config and imagemagick). And I got another error now: It seems like gtk is missing the **Adwaita theme**:

```
-------------------------------------------------------------------
You may only use the C/C++ Extension for Visual Studio Code
with Visual Studio Code, Visual Studio or Visual Studio for Mac
software to help you develop and test your applications.
-------------------------------------------------------------------
Breakpoint warning: No executable code of the debugger’s target code type is associated with this line.
Possible causes include: conditional compilation, compiler optimizations, or the target architecture of this line is not supported by the current debugger code type. - c:\Users\Febbe\workspace\xournalpp\src\core\gui\toolbarMenubar\PluginToolButton.h:22
Breakpoint warning: No executable code of the debugger’s target code type is associated with this line.
Possible causes include: conditional compilation, compiler optimizations, or the target architecture of this line is not supported by the current debugger code type. - c:\Users\Febbe\workspace\xournalpp\src\util\Stacktrace.cpp:65
Breakpoint warning: No executable code of the debugger’s target code type is associated with this line.
Possible causes include: conditional compilation, compiler optimizations, or the target architecture of this line is not supported by the current debugger code type. - c:\Users\Febbe\workspace\xournalpp\build-clang\_deps\fbbe_stacktrace-src\itanium\include\fbbe\stacktrace.h:312
Breakpoint warning: No executable code of the debugger’s target code type is associated with this line.
Possible causes include: conditional compilation, compiler optimizations, or the target architecture of this line is not supported by the current debugger code type. - c:\Users\Febbe\workspace\xournalpp\build-clang\_deps\fbbe_stacktrace-src\itanium\include\fbbe\stacktrace.h:325
Breakpoint warning: No executable code of the debugger’s target code type is associated with this line.
Possible causes include: conditional compilation, compiler optimizations, or the target architecture of this line is not supported by the current debugger code type. - c:\Users\Febbe\workspace\xournalpp\src\core\pdf\base\XojPdfPage.h:22
Loaded 'C:\Users\Febbe\workspace\xournalpp\build\xournalpp.exe'. Symbols loaded.
Loaded 'C:\Windows\System32\ntdll.dll'. 
Loaded 'C:\Windows\System32\kernel32.dll'. 
Loaded 'C:\Windows\System32\KernelBase.dll'. 
Loaded 'C:\Windows\System32\shell32.dll'. 
Loaded 'C:\Windows\System32\msvcp_win.dll'. 

```

**Comment by dg0yt** [comment]
You can build port `pkgconf` when you need a pkg-config implementation. (vcpkg uses msys2's pkg-config internally, but it is not exposed to users.)
Vcpkg focuses on libs, not tools. When do you need rsvg-convert now?
But this is build-time stuff. And unrelated to imagemagick. I don't see an imagemagick DLL, so I guess you use a static lib, with 3rd-party libs embedded into a "convenient" archive.

**Comment by Febbe** [comment]
> You can build port `pkgconf` when you need a pkg-config implementation. (vcpkg uses msys2's pkg-config internally, but it is not exposed to users.)
> Vcpkg focuses on libs, not tools. When do you need rsvg-convert now?
> But this is build-time stuff. And unrelated to imagemagick. I don't see an imagemagick DLL, so I guess you use a static lib, with 3rd-party libs embedded into a "convenient" archive.

rsvg-convert, imagemagick and pkg-config are all used at configure time, so I removed them from the path. Therefore none of their DLLs/libs should be pulled in. 
I cleaned the path, so that it only contained cmake,ninja and LLVM. Then I cleaned the vcpkg installation and recompiled everything.

The only static lib I use intentionally is the self compiled portaudiocpp, which dynamically links to portaudio.

When a static lib is the reason, it should be one of the vcpkg libs.

**Comment by Febbe** [comment]
I am not sure, but this looks like undefined behav. in the code of gtk3:

gdkselection-win32.c:1971
```c++
  info = g_slice_new (GdkWin32ClipboardQueueInfo);

  info->display = display;
  g_set_object (&info->requestor, requestor);
  info->selection = GDK_SELECTION_CLIPBOARD;
  info->target = target;
  info->idle_time = 0;
  info->time = time;
  info->action = action;
```

The debugger of windows initializes it with 0xbebebebe, when calling `g_set_object(&info->requestor, requestor)` it tries to unref 0xbebebebe, which is of course not an gobject.

**Comment by Febbe** [comment]
Just checked this in https://gitlab.gnome.org/GNOME/gtk/-/blob/gtk-3-24/gdk/win32/gdkselection-win32.c and they fixed that. @JonLiu1993 this issue is fixed, when you update gtk3 to the newest version.
